lib/SIL/SILVerifier.cpp

```diff
@@ -2675,14 +2675,26 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
               "no arguments");
   }
 
+  bool verifyBranchArgs(SILValue branchArg, SILArgument *bbArg) {
+    // NOTE: IRGen currently does not support the following method_inst
+    // variants as branch arguments.
+    // Once this is supported, the check can be removed.
+    require(
+        !isa<WitnessMethodInst>(branchArg) &&
+            !(isa<MethodInst>(branchArg) &&
+              cast<MethodInst>(branchArg)->getMember().getDecl()->isObjC()),
```

Comment:
@jckarter Addressed in https://github.com/apple/swift/pull/2508. I'll merge it once it passes the tests.